src/layerstack.stanza

```diff
@@ -576,49 +579,106 @@ doc:\<DOC>
-        prepreg
+        [copper-35um [prepreg prepreg2 prepreg3]]
         [copper-17_5um core]
       ]
   ```
   The LayerStack created has these layers:
+    ```
+     [  copper-35um
+       prepreg ; three adjacent prepreg layers
+       prepreg2
+       prepreg3
+       copper-17_5um
+       core
+       copper-17_5um
+       prepreg3
+       prepreg2
+       prepreg
+       copper-35um
+     ]
+    ```
+
+  Example 3: 6-layer stack with even number of layers, such as "JLC06201H-3313A"
+  ```
+    make-layer-stack("6-layer stack with even number of layers", outers, even-layers? = true) where :
+      val outers =
+        [
+          [copper-35um prepreg]
+          [copper-17_5um core]
+          [copper-17_5um prepreg2] ; => two layers of prepreg2 in the center
+        ]
+  ```
+  The LayerStack created has these layers:
+    ```
      [ copper-35um
        prepreg
-       prepreg ; two adjacent prepreg layers
        copper-17_5um
        core
        copper-17_5um
-       prepreg
+       prepreg2         ; two layers of prepreg2 in the center, not one single core/prepreg layer 
+       prepreg2         ;
+       copper-17_5um
+       core
+       copper-17_5um
        prepreg
        copper-35um
      ]
+    ```
 
   @param name The name of the LayerStack
-  @param top-layers The pairs of mechanical and dielectric layers, from outer to inner layers.
-  @param description The description of the LayerStack
+  @param top-layers A tuple of pairs of the mechanical layer and one dielectric layer or 
+    of the mechanical layer and a tuple of two or more dielectric layers, from outer to inner layers.
+  @param description (optional) The description of the LayerStack
+  @param soldermask (optional) The soldermask layer to be added as the outermost layer
+  @param even-layers? (optional) Whether there are an even number of layers. The default is `false`.
 <DOC>
-public defn make-layer-stack (name:String, outers:Tuple<[LayerSpec, LayerSpec]|LayerSpec>
-      -- description:String = ?, soldermask:LayerSpec = ?) -> LayerStack :
+public defn make-layer-stack (name:String,
+  outers:Tuple<[LayerSpec LayerSpec]|[LayerSpec Tuple<LayerSpec>]>
+      -- description:String = ?, 
+         soldermask:LayerSpec = ?,
+         even-layers?:True|False = false) -> LayerStack :
   val stack = #LayerStack(One(name), description)
   ;Access pairs of outer-layers in reverse order
   for outer in in-reverse(outers) do:
-    match(outer):
-      (outer:LayerSpec):
-        ; add one prepreg layer to top and bottom
-        add-top(outer, stack)
-        add-bottom(outer, stack)
-      ([metal, insul]:[LayerSpec LayerSpec]):
-        ; add [copper, prepreg] layers to top and bottom
+    ;Verify input
+    if outer[1] is Tuple<LayerSpec> and empty?(outer[1] as Tuple<LayerSpec>) :
+      throw(Exception("Invalid argument: The tuple of dialectric layers is empty. Expect [LayerSpec Tuple<LayerSpec>]. Found %_" % [outer]))
+    if not even-layers? :
```

Comment:
The `even-layers?` parameter here is making this function about 1000x more complicated than this needs to be.

Reply:
We need to remove this.